<div style="text-align:left;font-size:2em"><span style="font-weight:bolder;font-size:1.25em">SP2273 | Learning Portfolio</span><br><br><span style="font-weight:bold;color:darkred">Storing Data (Need) Exercises</span></div>

# Exercise 1 (Total recall?)

1. In both lists and arrays, order is very important.
   Both lists and arrays have the len() function.

2. Arrays insist on having only a single data type whereas lists are more accommodating.
An operation on a list works on the whole list whereas an operation on an array works on the individual elements of the array.


3. A dictionary holds data (values) paired with a key.


# Exercise 2 (Indexing)

In [14]:
py_list = ["a1", "b2", "c3", "d4", "e5", "f6", "g7", "h8", "i9", "j10"]
print(py_list[0])
print(py_list[2])
print(py_list[4])
print(py_list[6])
print(py_list[8])

a1
c3
e5
g7
i9


# Exercise 3 (Index again)

In [15]:
elements = ['Hydrogen',
            'Helium', 'Lithium',
            'Beryllium', 'Boron', 'Carbon',
            'Nitrogen', 'Oxygen',
            'Fluorine',
            'Neon']

In [19]:
print(elements[4])

Boron


In [20]:
print(elements[-6])

Boron


# Exercise 4 (How many ones)

In [21]:
numbers=[45, 60, 1, 30, 96, 1, 96, 57, 16, 1,
        99, 62, 86, 43, 42, 60, 59, 1, 1, 35,
        83, 47, 34, 28, 68, 23, 22, 92, 1, 79,
        1, 29, 94, 72, 46, 47, 1, 74, 32, 20,
        8, 37, 35, 1, 89, 29, 86, 19, 43, 61]                  

In [46]:
import numpy as np
np_numbers=np.array(numbers)
ones=np_numbers[np_numbers==True]
number_of_ones=ones.sum()
print(number_of_ones)

9


# Exercise 5 (A Matter of Statistics)

In [55]:
x_1 = [1, 4, 9, 16, 25, 36, 49, 64, 81, 100]
x_2 = [1, 8, 27, 64, 125, 216, 343, 512, 729, 1000]

## 1. Determine the means of x1 and x2

In [59]:
print(np.mean(x_1))

38.5


In [60]:
print(np.mean(x_2))

302.5


## 2. Determine the standard deviations of x1 and x2

In [61]:
print(np.std(x_1))

32.41990129534635


In [62]:
print(np.std(x_2))

326.0893282522444


## 3. Determine the Pearson correlation coefficient r for the two datasets

In [76]:
numerator_corrcoef=np.sum((x_1-np.mean(x_1))*(x_2-np.mean(x_2)))
denominator_corrcoef=np.sqrt(np.sum((x_1-np.mean(x_1))**2))*np.sqrt(np.sum((x_2-np.mean(x_2))**2))
corrcoef=numerator_corrcoef/denominator_corrcoef
print(corrcoef)

0.9871796626270701


## 4. Compare your results to np.corrcoef()

In [71]:
print(np.corrcoef(x_1,x_2))

[[1.         0.98717966]
 [0.98717966 1.        ]]


Both have 0.98717966
The manual calculation don't have 1 whereas the np calculation have 1.
The manual calculation is a float whereas the np calculation is a 2D list.